### DISCUSSION TITLE
HTTP to HTTPS redirection ?

### DISCUSSION THREAD

**Discussion by dev31337** [cesanta/mongoose#2400] [Ask questions, get answers] [answered]
Hi,

Is that possible to make mongoose do a HTTP to HTTPS redirection (on the same port) with one single (HTTPS) listener ?

If not, how can I do it since you can not have a HTTP listener and a HTTPS listener on the same port ?

Regards.

**Comment by scaprile** [comment, marked as answer]
That you do at application level. I don't think that can be done at the same port, but nobody prevents you from achieving that.
BTW, just searching for "redirect" sends you to #2183 , that already answered a similar question

**Comment by dev31337** [comment]
Thank you for your reply.

Sorry, but technically this is not the same question, because #2183 just wants to redirect a HTTP connection received on one port (ie: 80) to another port as HTTPS (ie: 443). (which is quite easy to do with two listeners)

Personally, I want to listen only to HTTPS (ie: 443) and if there is a HTTP request on this port, I want to be able to redirect it (using a 30X response) to the same port but in HTTPS.

**Reply by scaprile** [reply]
#2401 https://github.com/cesanta/mongoose/issues/2401#issuecomment-1742727849
